From 63ec66eeb1aefb7b0586a5eee3c0d0288e3451d3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fabian=20Gr=C3=BCnbichler?= <f.gruenbichler@proxmox.com>
Date: Wed, 11 Oct 2017 14:33:14 +0200
Subject: [PATCH 08/11] zfsutils-linux: install new files
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Fabian Grünbichler <f.gruenbichler@proxmox.com>
---
 debian/zfsutils-linux.install | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/debian/zfsutils-linux.install b/debian/zfsutils-linux.install
index 5b7dabfa9..b355ec47e 100644
--- a/debian/zfsutils-linux.install
+++ b/debian/zfsutils-linux.install
@@ -9,6 +9,8 @@ usr/lib/modules-load.d/ lib/
 lib/udev/
 etc/default/zfs
 etc/zfs/zfs-functions
+etc/zfs/zpool.d/
+usr/lib/zfs-linux/zpool.d/
 sbin/fsck.zfs
 sbin/mount.zfs
 sbin/ztest
@@ -19,6 +21,7 @@ sbin/zdb
 sbin/zpios
 sbin/zpool
 sbin/zfs
+usr/bin/zgenhostid
 usr/sbin/arc_summary
 usr/sbin/arcstat
 usr/sbin/dbufstat
@@ -32,3 +35,4 @@ usr/share/man/man8/zdb.8
 usr/share/man/man8/vdev_id.8
 usr/share/man/man8/mount.zfs.8
 usr/share/man/man8/fsck.zfs.8
+usr/share/man/man8/zgenhostid.8
-- 
2.14.1

