Index: pkg-zfs/debian/rules
===================================================================
--- pkg-zfs.orig/debian/rules
+++ pkg-zfs/debian/rules
@@ -112,12 +112,12 @@ override_dh_installinit:
 
 	@# Install the ZED init file.
 	dh_installinit -pzfsutils --onlyscripts --name=zfs-zed \
-	  --no-restart-on-upgrade --no-start -- start 07 S .       stop 08 0 1 6 .
+	  --no-restart-on-upgrade --no-start
 
 	@# Install the /etc/init.d/zfs-import script.
 ifeq ($(LSB_DISTRIBUTOR),Debian)
 	dh_installinit -pzfsutils --onlyscripts --name=zfs-import \
-	  --no-restart-on-upgrade --no-start -- start 07 S .       stop 07 0 1 6 .
+	  --no-restart-on-upgrade --no-start
 else
 	dh_installinit -pzfsutils --onlyscripts --name=zfs-import \
 	  --no-restart-on-upgrade --no-start
@@ -127,7 +127,7 @@ endif
 ifeq ($(LSB_DISTRIBUTOR),Debian)
 	@# Debian runs local mounts at sysv sequences [10..12] [08..09].
 	dh_installinit -pzfsutils --onlyscripts --name=zfs-mount \
-	  --no-restart-on-upgrade --no-start -- start 02 2 3 4 5 . stop 06 0 1 6 .
+	  --no-restart-on-upgrade --no-start
 else
 	dh_installinit -pzfsutils --onlyscripts --name=zfs-mount \
 	  --no-restart-on-upgrade --no-start
@@ -137,7 +137,7 @@ endif
 ifeq ($(LSB_DISTRIBUTOR),Debian)
 	@# Debian runs nfs-kernel-server at sysv sequence 17 01.
 	dh_installinit -pzfsutils --onlyscripts --name=zfs-share \
-	  --no-restart-on-upgrade --no-start -- start 27 2 3 4 5 . stop 05 0 1 6 .
+	  --no-restart-on-upgrade --no-start
 else ifeq ($(LSB_DISTRIBUTOR),Ubuntu)
 	@# Ubuntu runs nfs-kernel-server at sysv sequence 20 80.
 	@# iscsitarget at 20 20, and samba through upstart.
