Index: new-zfs/debian/zfsutils.zfs.default
===================================================================
--- new-zfs.orig/debian/zfsutils.zfs.default	2015-04-12 14:06:48.000000000 +0200
+++ new-zfs/debian/zfsutils.zfs.default	2015-04-14 07:59:27.000000000 +0200
@@ -36,15 +36,6 @@
 # Include '-o' for each option wanted.
 MOUNT_EXTRA_OPTIONS=""
 
-# Build kernel modules with the --enable-debug switch?
-ZFS_DKMS_ENABLE_DEBUG='no'
-
-# Build kernel modules with the --enable-debug-dmu-tx switch?
-ZFS_DKMS_ENABLE_DEBUG_DMU_TX='no'
-
-# Keep debugging symbols in kernel modules?
-ZFS_DKMS_DISABLE_STRIP='no'
-
 # Wait for this many seconds in the initrd pre_mountroot?
 # This delays startup and should be '0' on most systems.
 ZFS_INITRD_PRE_MOUNTROOT_SLEEP='0'
