diff --git a/debian/patches/series b/debian/patches/series
index ffafe6c..9823273 100644
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -1,12 +1,12 @@
 0002-Prevent-manual-builds-in-the-DKMS-source.patch
 0005-Remove-userland-dist-rules.patch
 libzfs-dependencies
-PR1099.patch
-PR1476.patch
+#PR1099.patch
+#PR1476.patch
 PR1867.patch
 PR2668.patch
-PR2790.patch
-PR3238.patch
+#PR2790.patch
+#PR3238.patch
 PR3465.patch
 PR3559.patch
 PR3560.patch
